contribute/design/1.x/upgrade/volume-pools-upgrade.md

```diff
@@ -0,0 +1,499 @@
+  look at the upgrade task (consider it as a claim) and create a 
+  specific upgrade task and bind it. Then the specific upgrade 
+  operator will operate on the their own resources.  
+
+- Should the UpgradeTask have a provision to perform upgrades on 
+  multiple resources of the same kind. For example, can a list of 
+  jiva volumes be specified in a single UpgradeTask. Adding multiple
+  jobs will result in adding additional status objects, which in 
+  turn will make the co-relation of the object to its result harder
+  to get. UpgradeTask CR provides a basic building block for 
+  constructing higher order automation. In the future, a BulkUpgrade
+  CR can be launched that can take multiple objects - probably 
+  even based a namespace group or pv labels etc. The controller
+  watching the BulkUpgrade can then launch individual UpgradeTasks. 
+
+   
+- How does this design compared to the `kubectl` based upgrade
+  introduced for upgrading from [0.8.2 to 0.9](https://github.com/openebs/openebs/tree/master/k8s/upgrades/0.8.2-0.9.0). 
+
+  The current design proposed in this document builds on top of the
+  0.8.2 to 0.9 design, by improving on usability and agility of the 
+  upgrade code development. 
+
+  The implementation made use of the CAS Template feature and comprised 
+  of the following: 
+  - Each type of upgrade - like jiva volume upgrade, cstor pool upgrade, 
+    were converted from shell to an CAS Template, with a series of  
+    upgrade tasks (aka runtasks) to be executed. 
+  - A custom resource called upgraderesults was defined to capture
+    the results while executing the run tasks.
+  - An upgrade agent, that can read the upgrade CAS Template and 
+    execute the run tasks on a given volume or pool.
+  - Administrator will have to create a Kubernetes Job out of the
+    upgrade agent - by passing the upgrade cas template and the object
+    details via a config map. 
+
+  While the above approach was good enough to be automated via `kubectl`, 
+  the steps that can be executed as part of upgrade were limited
+  by the features or constructs available within the CAST/RunTasks. 
+
+  Another short-coming of the CAST/RunTasks is the lack of developer
+  friendly constructs for rapidly writing new CAST/RunTasks. This was
+  severely impacting the time taken to push out new releases. 
+
+### High Level Design
+
+#### Upgrade Job Example
+
+Here is an example of Kubernetes Job spec for upgrading the jiva volume.
+```
+#This is an example YAML for upgrading jiva volume. 
+#Some of the values below needs to be changed to
+#match your openebs installation. The fields are
+#indicated with VERIFY
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  #VERIFY that you have provided a unique name for this upgrade job.
+  #The name can be any valid K8s string for name. This example uses
+  #the following convention: jiva-vol-<flattened-from-to-versions>-<pv-name>
+  name: jiva-vol-100110-pvc-713e3bb6-afd2-11e9-8e79-42010a800065
```

Comment:
Can we avoid even writing this standard i.e. VERIFY.
Let it be the job of higher order components.

Reply:
Added their as a comment for example.